src/NNLSSolverInterface.jl

```diff
@@ -0,0 +1,151 @@
+module NNLSSolverInterface
+
+export NNLSSolver
+
+using NNLS: QPWorkspace, load!, solve!
+importall MathProgBase.SolverInterface
+
+immutable NNLSSolver <: AbstractMathProgSolver
+end
+
+type NNLSModel <: AbstractLinearQuadraticModel
+    workspace::QPWorkspace{Float64, Int}
+    lb::Vector{Float64}
+    ub::Vector{Float64}
+    A::Matrix{Float64}
+    constr_lb::Vector{Float64}
+    constr_ub::Vector{Float64}
+    Q::Matrix{Float64}
+    q::Vector{Float64}
+    sense::Symbol
+    solution::Vector{Float64}
+    duals::Vector{Float64}
+
+    function NNLSModel()
+        m = new(QPWorkspace{Float64, Int}(0, 0))
+        resize!(m, 0, 0)
+        m
+    end
+end
+
+function Base.resize!(m::NNLSModel, q::Integer, n::Integer)
+    resize!(m.workspace, q, n)
+    m.lb = zeros(n)
+    m.ub = zeros(n)
+    m.A = zeros(q, n)
+    m.constr_lb = zeros(q)
+    m.constr_ub = zeros(q)
+    m.Q = zeros(n, n)
+    m.q = zeros(n)
+    m.sense = :Min
+    m.solution = fill(NaN, n)
+    m.duals = fill(NaN, q)
+end
+
+LinearQuadraticModel(s::NNLSSolver) = NNLSModel()
+
+numvar(work::QPWorkspace) = size(work.G, 2)
+numconstr(work::QPWorkspace) = size(work.G, 1)
+
+numvar(m::NNLSModel) = numvar(m.workspace)
+numconstr(m::NNLSModel) = numconstr(m.workspace)
+
+function loadproblem!(m::NNLSModel, A, lb, ub, obj, constr_lb, constr_ub, sense)
+    q, n = size(A)
+    if (q != size(m.A, 1)) || (n != size(m.A, 2))
+        resize!(m, q, n)
+    end
```

Comment:
So currently, the status will get set to `:Unsolved` in `loadproblem!` only if the problem is resized. Maybe set it to `:Unsolved` here explicitly? I guess it's kind of conflating the status of the `NNLSModel` and the status of the `NNLSQPWorkspace`; not sure what the right thing to do is here.